sambamba/pileup.d

```diff
@@ -424,8 +428,12 @@ class ChunkDispatcher(ChunkRange) {
 
     void queueResult(size_t num, char[] data) {
         synchronized(queue_mutex_) {
+            while(result_queue_.length > max_queue_length_) {
```

Comment:
I gave it a test run, and it stuck here. The condition should also have a clause `|| num == curr_num_`, otherwise the next chunk to output may arrive too late.